### PR TITLE
fix bug preventing queries with multiple colons, resolves #26

### LINKED ISSUE #26: search module behavior
there are inconsistent lucene queries involving urls and any other string that has a ':' in it. Currently you are only able to do a full string search, attempting to do the following does not work q:http\://da* or any kind of variation of it

### PR DESCRIPTION
**Pull request by jsnoble**
Example of using queries with colons

Note how querying the ipv6 is different, no *

ES 2.x
mapping: url => string, ipv6 => string
http://localhost:8000/api/v1/logstash?q=url:https://*
http://localhost:8000/api/v1/logstash?q=url:https://d*
http://localhost:8000/api/v1/logstash?q=ipv6:02ea:d*
http://localhost:8000/api/v1/logstash?q=ipv6:02e*

ES 5.x
mapping: url => string, ipv6 => url
http://localhost:8000/api/v1/logstash?q=url:https://*
http://localhost:8000/api/v1/logstash?q=url:https://d*
http://localhost:8000/api/v1/logstash?q=ipv6:027c:a741:6e0a:8185:6eb1:5ccf:b337:30fc
http://localhost:8000/api/v1/logstash?q=ipv6:027c:a741::/9

### CODE REVIEW

**Comment by jsnoble** [comment]
TLDR: ipv6 I can get to work with wildcards on 2.x and ip syntax on 5.x, however URLs are a hot mess, they only work with exact name queries. Wildcards only seem to work with ipv6 despite the mappings for url and ipv6 are the same

ES 2.x
URLS
    Failures
- http://localhost:9200/test2/_search?q=url:https://*&size=10000
- http://localhost:9200/test2/_search?q=url:https\://*&size=10000  => some how gave me everything
- http://localhost:9200/test2/_search?q=url:"https://*"&size=10000
- http://localhost:9200/test2/_search?q=url:"https\://*"&size=10000
- http://localhost:9200/test2/_search?q=url:\"https://*\"&size=10000
- http://localhost:9200/test2/_search?q=url:\"https\://*\"&size=10000

http://localhost:9200/test2/_search?q=url:https*&size=10000 => works right but anything with another colon does not

working:
 - http://localhost:9200/test3/_search?q=url:https\://dean.info&size=10000
 - http://localhost:9200/test3/_search?q=url:"https://dean.info"&size=10000
 - http://localhost:9200/test3/_search?q=url:"https\://dean.info"&size=10000

ipv6:
Appears to work
http://localhost:9200/test2/_search?q=ipv6:020d\:*&size=10000 => seems to work
http://localhost:9200/test2/_search?q=ipv6:020d\:03f3\:*&size=10000 => seems to work
http://localhost:9200/test2/_search?q=ipv6:020*&size=10000 => seems to work


ES 5.x

ipv6:
http://localhost:9200/test3/_searchq=ipv6:027c\:a741\:6e0a\:8185\:6eb1\:5ccf\:b337\:30fc&size=10000  => works
http://localhost:9200/test3/_search?q=ipv6:"027c:a741::/9"&size=10000 => works
http://localhost:9200/test3/_search?q=ipv6:"027c\:a741::/9"&size=10000 => works
http://localhost:9200/test3/_search?q=ipv6:"027c\:a741\:\:/9"&size=10000 => works

url:
Same as 2.x successes and errors
